Update jostick.cpp

gsbang97 opened the pull request:
조이스틱 완료
그전에는
1. 한줄을 모두 읽은 경우
2. 오른쪽에 가장 긴 A를 만나고 다시 왼쪽으로 읽은 경우
3. 왼쪽으로 가다가 긴 A를 만나고 오른쪽으로 읽는 경우
세가지 중에서 가장 적은 경우를 사용했었는데
1번에서 한줄에서 왼쪽(이경우 2번째 문자부터)이나 오른쪽에 A가 나열되어있는 경우 끝가지 탐색할 필요가 없는 경우를 고려해야 해서 해당 경우를 해결하고 난 뒤에 통과했습니다.
그리디? 라고 하기는 좀 그런문제인것 같네요.

gsbang97 committed:
Update jostick.cpp

조이스틱 완료

minji0401 reviewed (approved):
오 .. 금방해결하셨네요

gsbang97 committed:
Create 달나라토끼.cpp

7까지는 데이터를 입력해주고 그 뒤에 데이터에는 그 가격의 1,2,5,7원 적은 데이터중에 가장 작은 데이터에 1을 더해 저장했습니다.

gsbang97 committed:
Create 창영이커피.py

c++로 하기에는 좀 복잡하게 생각해서 파이썬으로 완성했네요..

gsbang97 commented:
4주차 과제도 commit되어있습니다.~~~

chhs2131 commented on giseung/week04/창영이커피.py:
```diff
@@ -0,0 +1,31 @@
+      min_num = [101, -1]
+      for c in coffees:
+        if k-c > 0 and numbers[k-c][0] != -1:
+          if min_num[0] > numbers[k-c][0]:
+            if c in numbers[k-c][1]:
+              min_num[0] = numbers[k-c][0]
+              min_num[1] = c
```
수량과 값을 동시에 관리해도 되겠네요..!  
기본 함수인 min 사용해 보시는 것도 좋을 것 같아요!

gsbang97 replied:
아 맞네요 ! 파이썬에 min이 있는걸 깜빡했네요 ㅎㅎㅎ

gsbang97 committed:
Create 쉬운계단수.cpp

DP로 어떻게 푸나 싶어서 질문하기에 있는 것을 좀 참고했습니다.
stair[a][b] = a자리수의 b 숫자로 끝나는 수의 개수 로 저장하여 사용했습니다. 다들 이 방식으로 사용하더라고요
실제로 코테하면 이런 생각 어떻게 하나 싶습니다.
